assets/js/frontend/course-video/youtube-extension.js

```diff
@@ -0,0 +1,58 @@
+( () => {
+	const tag = document.createElement( 'script' );
+	tag.id = 'iframe-api';
+	tag.src = 'https://www.youtube.com/iframe_api';
+	const firstScriptTag = document.getElementsByTagName( 'script' )[ 0 ];
+	firstScriptTag.parentNode.insertBefore( tag, firstScriptTag );
+
+	window.onYouTubeIframeAPIReady = function () {
+		document
+			.querySelectorAll(
+				'.sensei-course-video-youtube-container > iframe'
+			)
+			.forEach( ( element ) => {
+				const player = new YT.Player( element, {
+					events: {
+						onStateChange: ( event ) => {
+							const playerStatus = event.data;
+
+							if (
+								window.videoBasedCourseDisableCompleteButton !==
+									undefined &&
+								window.videoBasedCourseDisableCompleteButton &&
+								playerStatus === YT.PlayerState.ENDED
+							) {
+								window.videoBasedCourseDisableCompleteButton = false;
+							}
+
+							if (
+								window.videoBasedCourseAutoComplete !==
+									undefined &&
+								window.videoBasedCourseAutoComplete &&
+								playerStatus === YT.PlayerState.ENDED
+							) {
+								// submit complete lesson form
+								document
+									.querySelectorAll( '.lesson_button_form' )
+									.forEach( ( form ) => {
+										const action = form.querySelector(
+											'input[name=quiz_action]'
+										).value;
+										if ( action !== 'lesson-complete' ) {
+											return true;
+										}
+										form.submit();
+									} );
+							}
+						},
+					},
+				} );
+				if ( window.videoBasedCourseAutoPause ) {
+					// eslint-disable-next-line @wordpress/no-global-event-listener
+					window.addEventListener( 'blur', () => {
```

Comment:
I was trying to attach my handler to the window inside iframe:
```
element.contentWindow.addEventListener( 'blur', () => {
    player.pauseVideo();
} );
```
But I got an error:
```
Uncaught DOMException: Blocked a frame with origin "http://localhost:8889" from accessing a cross-origin frame.
    at eval (webpack-internal:///./js/frontend/course-video/youtube-extension.js:40:31)
    at NodeList.forEach (<anonymous>)
    at window.onYouTubeIframeAPIReady (webpack-internal:///./js/frontend/course-video/youtube-extension.js:9:82)
    at https://www.youtube.com/s/player/8da38e9a/www-widgetapi.vflset/www-widgetapi.js:958:151
    at https://www.youtube.com/s/player/8da38e9a/www-widgetapi.vflset/www-widgetapi.js:958:201
```
I wonder if there is any workaround for that?